kernel/src/main/scala/cats/kernel/Hash.scala

```diff
@@ -0,0 +1,52 @@
+package cats.kernel
+
+import scala.{specialized => sp}
+import scala.util.hashing.Hashing
+
+/**
+ * A type class used to represent a hashing scheme for objects of a given type.
+ * For any two instances `x` and `y` that are considered equivalent under the
+ * equivalence relation defined by this object, `hash(x)` should equal `hash(y)`.
+ * @author Tongfei Chen
+ */
+trait Hash[@sp A] extends Any with Eq[A] with Serializable { self =>
+
+  /**
+   * Returns the hash code of the given object under this hashing scheme.
+   */
+  def hash(x: A): Int
```

Comment:
64 bit JVMs store both `Int` and `Long` as 64 bit numbers, so there is no advantage to use `Int` for space preserving reasons. However if you want to make use of existing `hashCode` algorithms that return `Int` then that would be a good reason to keep it that way.